assemblies-valid/AssembliesValid.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using System.Runtime;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Xunit;
+
+namespace AssembliesValid
+{
+    public class AssembliesValid
+    {
+        // https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/coreclr/src/inc/pedecoder.h#L90
+        public static readonly int IMAGE_FILE_MACHINE_NATIVE_OS_OVERRIDE_LINUX = 0x7B79;
+
+        // https://github.com/dotnet/runtime/blob/master/src/coreclr/src/inc/corcompile.h
+        public static readonly int CORCOMPILE_SIGNATURE = 0x0045474E; // 'NGEN'
+
+        // https://github.com/dotnet/runtime/blob/master/src/coreclr/src/inc/readytorun.h
+        public static readonly int READYTORUN_SIGNATURE = 0x00525452; // 'RTR'
+
+        public static string[] FileNameBlacklist =
+        {
+            "mscorlib.dll",
+            "msdia140.dll",
+            "System.Private.CoreLib.dll",
+            "System.Runtime.WindowsRuntime.dll",
+        };
+
+        public static Regex[] PathBlacklist =
+        {
+            new Regex(".*\\.resources\\.dll$"),
+            new Regex("/sdk/"),
+            new Regex("/packs/"),
+        };
+
+        [Fact]
+        public void ValidateAssemblies()
+        {
+            string dotnetPath = null;
+            int exitCode = RunProcessAndGetOutput(new string[] { "command", "-v", "dotnet" }, out dotnetPath);
+            if (exitCode != 0)
+            {
+                Console.Error.WriteLine("'dotnet' command not found");
```

Comment:
This test is a Console application in disguise :smile: 
General practice is not to do use `Console` in tests, and split them into different `[Fact]`/`[Theory]`. The `[Fact]` that failed, and the exception thrown should be enough to figure out what is failing.

I'm fine if you want to keep the test as is.